### DISCUSSION TITLE
Improve github search

### DISCUSSION THREAD

**Discussion by pakirby1** [community/community#4436] [Other Feature Feedback, Questions, & Ideas]
Currently the search functionality for github could provide a better experience, see this for more info…

https://github.com/isaacs/github/issues/908#issuecomment-871539638

**Comment by TPS** [comment]
# I know this is different than issues, but discussions _really_ rely on upvotes (**↑**) than thumbsup (👍🏾).  Could all y'all who 👍🏾ed go back & **↑**?

**Reply by MichaelAkvo** [reply]
That's very confusing :/ What's the point of the thumbs up then?

**Comment by DiegoFleitas** [comment]
I don't even know if this a limitation of the search or a bug, but looks like a bug to me...

![image](https://user-images.githubusercontent.com/15369935/139945030-89dca9a4-2c05-4e20-b0ff-f34f9f853937.png)


https://github.com/serverless/serverless/blob/7bb2520f491d008075eb08be64bdb2e0b60ec5c6/lib/plugins/create/templates/aws-nodejs-typescript/src/functions/hello/handler.ts#L3

**Comment by Quezion** [comment]
Searched a URL today -- first search displayed 0 results.

Second search _for the same string_ showed 1 Code result, but still said `0` in all the leftside columns.

Can GitHub dump plaintext rgrep results? That or cut the feature & advise users search code locally -- because the current state of search is degrading for _the_ mainstream code platform in 2022.

**Comment by TPS** [comment]
Heck, even just wiring Bing into this might be better.

**Comment by adrielldagasuan** [comment]
Same thing for me. It feels like sometimes you have to "prime" search by just doing multiple iterations, which is a pain and also not good for the rate limits.

**Comment by MichaelAkvo** [comment]
Are the alternatives to github's search? Searching nixpkgs for code barely ever works https://github.com/NixOS/nixpkgs/search?q=vim_configurable

**Reply by MichaelAkvo** [reply]
OK, found 2 that had been recommended to me a while back

https://grep.app/search
https://sourcegraph.com/search

**Comment by davidstaheli** [comment]
A technology preview of GitHub's upcoming code search can be found at https://cs.github.com/.

**Comment by xiaohk** [comment]
The fuzzy search on the new code search is still bad.

For example, searching `cv2.Laplacian(img, 3)` gives 3 result while `cv2.Laplacian(img,3)` gives 0 results...

![image](https://user-images.githubusercontent.com/15007159/202089749-e7e14f64-61e8-4d62-957d-d014f886008d.png)

![image](https://user-images.githubusercontent.com/15007159/202089687-c6f2782e-ebef-46b9-9bf8-19042cfee7dc.png)

**Comment by chriscz** [comment]
Partial text never matches in issue search to this day. I tend to go to Google and do a `url:GITHUB_REPO` to find similar issues.
